opensearch-operator/api/v1/opensearch_types.go

```diff
@@ -43,17 +43,16 @@ type GeneralConfig struct {
 }
 
 type NodePool struct {
-	Component    string              `json:"component"`
-	Replicas     int32               `json:"replicas"`
-	DiskSize     int32               `json:"diskSize,omitempty"`
-	Cpu          int32               `json:"cpu,omitempty"`
-	Memory       int32               `json:"memory,omitempty"`
-	Jvm          string              `json:"jvm,omitempty"`
-	Roles        []string            `json:"roles"`
-	Tolerations  []corev1.Toleration `json:"tolerations,omitempty"`
-	NodeSelector map[string]string   `json:"nodeSelector,omitempty"`
-	Affinity     *corev1.Affinity    `json:"affinity,omitempty"`
-	Persistence  *PersistenceConfig  `json:"persistence,omitempty"`
+	Component    string                      `json:"component"`
+	Replicas     int32                       `json:"replicas"`
+	DiskSize     int32                       `json:"diskSize,omitempty"`
```

Comment:
To be consistent I think this should be a string so that users can specify something like `20Gi`

Reply:
Hey @swoehrl-mw I have not modified anything related to disk, my PR is for cpu and memory reconciler plus able to scale dashboard pods based on user entered replicas. If need to modify Persistence to string how about I create a new issue and push a new PR, this PR is for 
https://github.com/Opster/opensearch-k8s-operator/issues/71
https://github.com/Opster/opensearch-k8s-operator/issues/68

Reply:
good idea, please open an issue on that